arelle/ModelRelationshipSet.py

```diff
@@ -344,6 +342,9 @@ def label(self, modelFrom, role, lang, returnMultiple=False, returnText=True, li
                 labels = (labelsHintedLink or labelsDefaultLink or labelsOtherLinks)
         if len(labels) > 1: # order by priority (ignoring equivalence of relationships)
             labels = sorted(labels, key=lambda rel: rel.priority, reverse=True)
+        # (lang, labels) where labels is list[str] if returnText else list[ModelObject]
+        shorter: tuple[str, list[Any]] | None = None
+        longer: tuple[str, list[Any]] | None = None
```

Comment:
A small dataclass or namedtuple would be preferable to avoid having to keep track of index meaning.

```python
@dataclass
class LangLabels:
    lang: str
    labels: list[ModelObject] | list[str]

longer = LangLabels(labelLand, [text])
len(longer.lang) > len(labelLang)
longer.labels.append(text)
```

or:
```python

```